pynws/raw_data.py

```diff
@@ -1,22 +1,29 @@
 """Functions to retrieve raw data."""
 from datetime import datetime
 import logging
+from typing import Any, Dict, Optional
+from aiohttp import ClientSession
 
 from . import urls
 from .const import API_ACCEPT, API_USER
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def get_header(userid):
+def get_header(userid: str) -> Dict[str, str]:
     """Get header.
 
     NWS recommends including an email in userid.
     """
     return {"accept": API_ACCEPT, "User-Agent": API_USER.format(userid)}
 
 
-async def _make_request(websession, url, header, params=None):
+async def _make_request(
+    websession: ClientSession,
+    url: str,
+    header: Dict[str, str],
+    params: Optional[Dict[str, Any]] = None,
+) -> Dict[str, Any]:
```

Comment:
Technically, this method could return any valid JSON type. See [this link](https://github.com/python/typing/issues/182#issuecomment-899624078) for a valid JSON annotation. Maybe this function needs to check if the result is a `dict` and fail if it doesn't.

Reply:
I'm now surprised that this passes the static type checking.  Maybe the `aiohttp` response only handles `dict` conversion?

Reply:
```
obs = await res.json()
```
It's because the `ClientResponse.json()` returns `Any`.

Reply:
Ah, I think you are right that if we keep this type annotation then it would be expected that this is controlled for.  Or we could type it as `Any`?

Reply:
All subsequent code assumes that it's a `Dict[str, Any]` so it would make sense to validate that somewhere.

Reply:
Fixed